server/src/providers/blockchain.ts

```diff
@@ -233,14 +233,20 @@ export class blockchain {
   * @return {Promise<void>}
   */
   async checkAddressesClaimed(arrayOfAddresses: string[]): Promise<void> {
-    const promises = arrayOfAddresses.map(async address => {
-      if (!(await AddressesClaimedProvider.checkIfExists(address))) {
-        await PropertyProvider.increment('totalUniqueClaimers')
-        await AddressesClaimedProvider.create({ address })
-      }
-    })
-
-    await Promise.all(promises)
+    // there could be duplicates, so need to get unique values
+    // new Set([...]) -> will return unique values from received array
+    const uniqueAddresses = [...new Set(arrayOfAddresses)]
+
+    // check multiple addresses exists by one db query
+    const { nonExisted } = await AddressesClaimedProvider.checkIfExistsMultiple(uniqueAddresses)
```

Comment:
the check and insert can be  done in one command using update with upsert:true and $setOnInsert
https://docs.mongodb.com/manual/reference/operator/update/setOnInsert/
and from the result read nUpserted to get the number of documents inserted